src/sections/provider/ProviderSelectModal.tsx

```diff
@@ -1,192 +1,97 @@
-import { PROVIDERS, useProviderRpcUrlStore } from "api/provider"
-import { Button } from "components/Button/Button"
+import { useProviderRpcUrlStore } from "api/provider"
 import { Modal } from "components/Modal/Modal"
-import { Separator } from "components/Separator/Separator"
-import { Fragment, useState } from "react"
+import { useState } from "react"
 
-import { SContainer, SHeader } from "./ProviderSelectModal.styled"
+import { Switch } from "components/Switch/Switch"
+import { Text } from "components/Typography/Text/Text"
 import { useTranslation } from "react-i18next"
-import { Controller, useForm } from "react-hook-form"
-import { ProviderInput } from "./components/ProviderInput/ProviderInput"
+import { ProviderSelectForm } from "sections/provider/ProviderSelectForm"
 import { useRpcStore } from "state/store"
-import { FormValues } from "utils/helpers"
-import { useMutation } from "@tanstack/react-query"
-import { connectWsProvider } from "./ProviderSelectModal.utils"
-import { ProviderItem } from "./components/ProviderItem/ProviderItem"
+import { SSwitchContainer } from "./ProviderSelectModal.styled"
 import { DeleteModal } from "./components/DeleteModal/DeleteModal"
-import { SubstrateApis } from "@galacticcouncil/xcm-core"
+import { Button } from "components/Button/Button"
 
 export function ProviderSelectModal(props: {
   open: boolean
   onClose: () => void
 }) {
-  const preference = useProviderRpcUrlStore()
-  const activeRpcUrl = preference.rpcUrl ?? import.meta.env.VITE_PROVIDER_URL
-  const [userRpcUrl, setUserRpcUrl] = useState(activeRpcUrl)
+  const { setRpcUrl, autoMode, setAutoMode } = useProviderRpcUrlStore()
   const [removeRpcUrl, setRemoveRpcUrl] = useState<string | undefined>()
   const { t } = useTranslation()
-  const { rpcList, addRpc, removeRpc } = useRpcStore()
-
-  const form = useForm<{ address: string }>({
-    defaultValues: { address: "wss://" },
-    mode: "onChange",
-  })
-
-  const mutation = useMutation(async (value: FormValues<typeof form>) => {
-    try {
-      const provider = await connectWsProvider(value.address)
-
-      const apiPool = SubstrateApis.getInstance()
-      const api = await apiPool.api(provider.endpoint)
-
-      const relay = await api.query.parachainSystem.validationData()
-      const relayParentNumber = relay.unwrap().relayParentNumber
-
-      if (relayParentNumber.toNumber()) {
-        addRpc(value.address)
-        form.reset()
-      }
-    } catch (e) {
-      if (e === "disconnected")
-        form.setError("address", {
-          message: t("rpc.change.modal.errors.notExist"),
-        })
-      throw new Error(t("rpc.change.modal.errors.notExist"))
-    }
-  })
+  const { removeRpc } = useRpcStore()
 
   return (
     <>
       <Modal
         open={props.open}
         onClose={props.onClose}
         title={t("rpc.change.modal.title")}
+        headerVariant="FontOver"
       >
-        <>
-          {import.meta.env.VITE_ENV !== "production" && (
```

Comment:
Have you removed that conditional?

Reply:
Yes, user can add his own RPC even on production now.